client/src/components/App.jsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import TopBar from './TopBar.jsx'
+import VideoPLayer from './VideoPLayer.jsx'
+import StreamDetail from './StreamDetail.jsx'
+import '../styles.css';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      streamerId: 3,
+      streamerData: {}
+    }
+    this.commasWithNumber = this.commasWithNumber.bind(this)
+  }
+
+  componentDidMount() {
+    const streamerId = this.props.streamerId || this.state.streamerId
+    fetch('/api/livestream/' + streamerId)
+    .then(res => res.json())
+    .then(data => {
+      data[0].total_views = this.commasWithNumber(data[0].total_views)
+      data[0].watching_now = this.commasWithNumber(data[0].watching_now)
+      console.log(typeof data[0].video_url)
+      this.setState({
+        streamerId: data[0].id,
+        streamerData: data[0]
+      })
+    })
+  }
+
+  commasWithNumber(num) {
+    if (num) {
+      return num.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",")
+    }
+  }
+
+  render() {
+    return (
+    <div id='video-player-container'>
+      {this.state.streamerData &&
```

Comment:
I think streamerData is initialized to {}, which I think is truthy, so this is always true and can maybe be removed?

Reply:
The initialized obj doesn't have the properties until the data is fetched. It won't render properly without this conditional. I have something similar for the videopplayer (took like an hour to remember that.)